docs/source/tutorial_torch_generator_agent.md

```diff
@@ -154,3 +153,79 @@ parlai train_model -m examples/seq2seq \
 ```
 
 You should get a perplexity of around 140 and a token accuracy of around 28% on the ConvAI2 validation/test set.
+
+### Using custom loss function
+
+It is common in downstream tasks that people want to use a custom loss function on top of the original cross-entropy loss used to train the generative model. You can achieve this by overriding the `compute_loss` function. And a concrete example can be found [here](https://github.com/facebookresearch/ParlAI/blob/0c25349ac91abe2a5f232c63985aed14a37f5dcf/projects/wizard_of_wikipedia/generator/agents.py#L121).  Please note `compute_loss` changes its output type based on the `return_output` parameter. It either returns only the loss or returns the loss and whatever the model's `output.forward()` was. Be mindful of that. 
+
+Let us walk though an example:
+
+```python
+    def compute_loss(self, batch, return_output=False):
```

Comment:
instead of copy pasting, can you give a simple example?

Reply:
Can you make up an use case? I

Reply:
Label smoothing would be really nice, but you could do something silly like "Also minimize the sum of all the logits"

Reply:
how about a regularizer on the encoder states.

`loss = nll_loss + encoder_states.square().sum()`